database2/cassandraKS.js

```diff
@@ -0,0 +1,39 @@
+const cass = require('cassandra-driver');
+
+const client = new cass.Client({
+  contactPoints: ['localhost'],
+  localDataCenter: 'datacenter1',
+  keyspace: 'system'
+});
+
+const createProductsKeyspace = () => {
+  client.connect()
+    .then(() => {
+      const query = "DROP KEYSPACE similar_prod";
+      return client.execute(query);
+    })
+    .catch((err) => {
+      console.error('keyspace not defined or other error: ', err);
+    })
+    .then((result) => {
+      console.log('keyspace dropped: ', result);
+      const query = "CREATE KEYSPACE IF NOT EXISTS similar_prod WITH replication =" + "{'class': 'SimpleStrategy', 'replication_factor': '1'}";
+      return client.execute(query);
+    })
+    .then((result) => {
+      console.log('keyspace created: ', result);
```

Comment:
Nice way to ensure you have a clean slate every time you run your seed. I can't say I fully understand keyspaces yet, but does the fact that you're creating a keyspace have any weird interaction with the fact that you're connecting to the 'system' keyspace? I'm not pointing out an issue or providing any sort of constructive criticism here -- I genuinely don't know!

Reply:
Fair question! I found with both cassandra and postgres, you are required to connect with an existing keyspace, or database, respectively, in order to create any additional keyspace/database or table. I might be missing something, but with Cassandra it didn't cause any problems to then create a new keyspace from the client connection. It might make sense to disconnect and reconnect to client using the created keyspace before proceeding? In Cassandra, you can precede your tablename with your keyspace name when creating, in order to ensure it is created inside the correct keyspace (eg: to create the table "products" inside similar_prod, I used CREATE TABLE similar_prod.products. Definitely feel like I'm hodge-podging and JUST looked at this https://github.com/datastax/nodejs-driver/blob/master/examples/basic/basic-execute-flow.js which I plan to reference as I refactor!